### ISSUE TITLE
Revert to older chat system

### ISSUE THREAD

**Issue by The-PC-Tech-Guy** [SimpleServer/SimpleServer#214]
The problem with this new chat system is that it bypasses any server mod. The only option is to simply give a copy of the message to the server mod - SS still handles the message itself and sends it to the specified clients. This disallows any chat-modifying plugins such as CensorIt ( http://dev.bukkit.org/server-mods/censorit/ ) to work at all - if it does happen to modify a message, it is then resent from Bukkit itself, and as a new copy, causing duplicated posts.

I would like to see an option to revert to the old chat system, which I believe processed the messages after they passed through the server (since I was able to censor chat before (with a different plugin at the time) and still keep the SS group titles).

As a secondary request, I would like to have some type of chat censoring feature included in SS. The API of CensorIt is publicly available (see link), so this may be able to give you some help in creating one. [It is indeed a very great plugin that is more than just a word filter, it also can search for and detect any possible attempts to work around it, as well as an option to verify words with an online dictionary.]

**Comment by dola** [comment]
Yeah currently all chat is only processed by SimpleServer. I see that this introduces some issues with bukkin plugins that work with chat (especially censor plugins and others that change chatmessages before returning).

We could think about integrating the old style for public messages but if you have a look at the chatrooms of SimpleServer (that came with the new message system) it's kind of hard and especially bound to a lot of overhead if we would let all messages pass the server and then somehow filter the recipents when they're sent back from the server...

A censor extension for SimpleServer could be implemented, but I myself don't really see why at all one should censor chat messages... :P Are your users so childish that they don't even respect basic rules? And a solution that works with an external API would most likely be ways to slow for live integration. But the funcionality of CensorIt seems to be huge and powerful if I look at it :)

**Comment by The-PC-Tech-Guy** [comment]
Before, there was only the /local command, which was pretty much the only way to have direct SimpleServer chat. And yes, I do have some users who simply don't care about the rule (it's pretty much the only rule), but still, I'd rather have a filter - that's my preference. If SS had such a way, it will definitely keep me using it; I have considered many times to just go to Bukkit itself for a few reasons, but the features here on SS keep improving, with no fear of them breaking ;)

Update: Nevermind, just dropped SimpleServer. Sorry guys, I'll check back when things become simple again :P but it was a good ride. I'll probably use this for vanilla servers, as it is not really meant anywhere near for Bukkit servers, especially at where it is right now :/

**Comment by kevinpeno** [comment]
This wouldn't be a big issue if SimpleServer supported plugins itself. This would not only keep the core "simple" but would allow easy expansion on core features. I'm hitting this same stretch myself and am willing to form a coalition to fork off and create a plugin based version of the software.

**Comment by bryanburke** [comment]
A plugin system for SimpleServer would be nice. Several of the current features (like !myarea) could be moved to plugins to make the base wrapper more lightweight. Perhaps even the new chat system could be outsourced to a plugin.

**Comment by kevinpeno** [comment]
Just to clarify my position in the last reply. I'm not trying to form a coup and new product. I am just willing to get a group of people willing to experiment with the idea as a separate branch and, hopefully, get it merged back in once it seems acceptable.

**Comment by heapoutofspace** [comment]
As you might have noticed dola and I stopped actively developing SimpleServer some time ago. If you (@kevinpeno and @bryanburke) are willing to build a plugin system, or actually any other addition to the current software, I'll gladly grant you access to the SimpleServer organization. If you want to discuss anything, we're in our IRC Channel almost all day ;)

**Comment by bryanburke** [comment]
Could you PM us with the IRC channel? Also, don't forget about @ravngr, as he also seems to have an interest in the project.

**Comment by heapoutofspace** [comment]
I think the channel is in the README or at least the forum. But it's #SimpleServer on irc.euirc.net if you're too lazy ;)

I already contacted @ravngr and added him to the contribution team.

**Comment by kevinpeno** [comment]
@sadimusi I'd be glad to help where I can. First place I'd probably attack would be organizing issues and, hopefully, updating the wiki.
